model/registry/geo.yaml

```diff
@@ -0,0 +1,114 @@
+groups:
+  - id: registry.geo
+    type: attribute_group
+    prefix: geo
+    brief: >
+      Geo fields can carry data about a specific location related to an event.
+      This geolocation information can be derived from techniques such as Geo IP, or be user-supplied.
+    attributes:
+      - id: city_name
+        stability: experimental
+        type: string
+        brief: >
+          City name.
+        examples: [ 'Montreal', 'Berlin' ]
+      - id: continent_code
+        stability: experimental
+        brief: >
+          Two-letter code representing continent’s name.
+        type:
+          allow_custom_values: false
+          members:
+            - id: 'af'
+              stability: experimental
+              value: 'AF'
+              brief: 'Africa'
+            - id: 'an'
+              stability: experimental
+              value: 'AN'
+              brief: 'Antarctica'
+            - id: 'as'
+              stability: experimental
+              value: 'AS'
+              brief: 'Asia'
+            - id: 'eu'
+              stability: experimental
+              value: 'EU'
+              brief: 'Europe'
+            - id: 'na'
+              stability: experimental
+              value: 'NA'
+              brief: 'North America'
+            - id: 'oc'
+              stability: experimental
+              value: 'OC'
+              brief: 'Oceania'
+            - id: 'sa'
+              stability: experimental
+              value: 'SA'
+              brief: 'South America'
+      - id: continent_name
+        stability: experimental
+        type: string
```

Comment:
should it be also an enum then?

Reply:
I took this from https://github.com/open-telemetry/semantic-conventions/pull/351 - so back then it seemed to be ok to have `continent_code` as enum and `continent_name` as string. At least I don't see any discussion or rejection for this. [In ECS both are `keyword`](https://www.elastic.co/guide/en/ecs/current/ecs-geo.html) - so we can't get inspired from ECS in this case.

What I have found is that `continent_code` was added afterwords: https://github.com/elastic/ecs/pull/1229 - although the motivation mentioned there doesn't really help to answer this question either.

I don't feel super strongly about this, but if we make this into an enum as well, then what's the point of having both `continent_code` and `continent_name` both as enum with fixed values? We'd have basically the same information, one would be the short code (like `eu`), the other one the full name (like `Europe`), but it'd be the exact same information. So if this becomes an enum as well, then my question is: what's the point having both and why don't we drop one?

If however this remains a string, then this will enable implementation to not parse the values into the fixed enum values and only use `continent_name` as string (just store it as-is without any parsing) and not use `continent_code`. Whether this is a good thing and we want to have this is still a question.

Reply:
For me enums are useful if we want to have a limited set of answers and also low cardinality - if we save it just as string without any parsing we could have different spelling for the same thing. For the fixed set, and continents and their name is a small fixed set I feel enum has the best fit. 

I could see that both enums are basically the same thing, but I could see the benefit depending on the specific use cases, where one is using code and another - name.

Reply:
I think this comes down to the question: what do we want to allow users when it comes to continent? 

- Force them to only have a small set of fixed values? (due to the benefits you mentioned) -> enum
- Allow them to save parsing (and as you say may have different spelling, etc) -> string

> I could see that both enums are basically the same thing, but I could see the benefit depending on the specific use cases, where one is using code and another - name.

Do you think is this benefit worth the duplicated storage and the potential confusion of having 2 attributes with the same info?

The more I think about this the more I'd lean towards leaving `continent_name` as string. Also - that's more aligned with other fields: e.g. `country_name` is a string, but could be enum as well - of course that has more values, but still somewhat analogue to `continent_name`.

Reply:
I would say countries are more to have changes in its number/naming than continents but that's not really relevant for this PR.
I don't see the benefit of having it as string other than saving on parsing, but for small subset it's not a big deal.

Saying all this I don't have a strong opinion that it should be enum, if we decide to do otherwise.

Reply:
Thanks for the feedback - I also don't feel super strongly either, so enum is totally acceptable as well.

Let's see what others say. @rogercoll since you worked on this in the collector: any thoughts on this?

Reply:
I would also be tempted to use an enum for greater restriction on user input, but from a user perspective, the only unique field in any natural language seems to be the `continent_code`. Actually, while using the GeoIP database from MaxMind, you can directly retrieve the `continent_code` for any given an IP. However, if you want to retrieve the `continent_name`, you must specify a locale/name. See the [example](https://github.com/maxmind/MaxMind-DB/blob/main/source-data/GeoIP2-City-Test.json#L5).

In the [geoipprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/geoipprocessor) we are going to add the `continent_name` using the "english" name, but it makes me wonder the possibility of allowing the user to configure which name to use. In that case, being a string would be more suitable.

Reply:
> In the [geoipprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/geoipprocessor) we are going to add the continent_name using the "english" name, but it makes me wonder the possibility of allowing the user to configure which name/locale to use. In that case, being a string would be more suitable.

Yes, that's a good reason for it to be a string